src/libs/DateUtils.js

```diff
@@ -291,6 +291,105 @@ function getDateStringFromISOTimestamp(isoTimestamp) {
     return dateString;
 }
 
+/**
+ * @returns {string} example: 2023-05-16
+ */
+function getThirtyMinutesFromNow() {
+    return moment().add(30, 'minutes').format('YYYY-MM-DD HH:mm:ss');
+}
+
+function getOneHourFromNow() {
+    return moment().add(1, 'hours').format('YYYY-MM-DD HH:mm:ss');
+}
+
+/**
+ * @returns {string} example: 2023-05-16
+ */
+function getEndOfToday() {
+    return moment().endOf('day').format('YYYY-MM-DD HH:mm:ss');
+}
+
+/**
+ * @returns {string} example: 2023-05-16
+ */
+function getEndOfWeekFromNow() {
+    return moment().add(7, 'days').format('YYYY-MM-DD HH:mm:ss');
+}
+
+/**
+ * @param {string} time
+ * @returns {string} example: 2023-05-16
+ * it's kind of 'newer' date
+ */
+function extractDate(time) {
+    return moment(time).format('YYYY-MM-DD');
+}
+
+/**
+ * @param {string} dateTimeString
+ * @returns {string} example: 2023-05-16
+ * it's kind of 'newer' date
+ */
+function extractTime(dateTimeString) {
+    return moment(dateTimeString).format('hh:mm');
+}
+/**
+ * @param {string} dateTimeString
+ * @returns {string} example: 2023-05-16
+ * it's kind of 'newer' date
+ */
+function extractTime12Hour(dateTimeString) {
+    return moment(dateTimeString).format('hh:mm A');
+}
+
+/**
+ * @param {string} dateTimeString
+ * @returns {string} example: 2023-05-16
+ * it's kind of 'newer' date
+ */
+function formatDateTo12Hour(dateTimeString) {
+    if (!dateTimeString) return '';
+    return moment(dateTimeString).format('YYYY-MM-DD hh:mm A');
+}
+
+/**
+ * @param {string} type - one of the values from CONST.CUSTOM_STATUS_TYPES
+ * @returns {string} example: 2023-05-16
+ */
+function getDateBasedFromType(type) {
+    switch (type) {
+        case CONST.CUSTOM_STATUS_TYPES.THIRTY_MINUTES:
+            return getThirtyMinutesFromNow();
+        case CONST.CUSTOM_STATUS_TYPES.ONE_HOUR:
+            return getOneHourFromNow();
+        case CONST.CUSTOM_STATUS_TYPES.AFTER_TODAY:
+            return getEndOfToday();
+        case CONST.CUSTOM_STATUS_TYPES.AFTER_WEEK:
+            return getEndOfWeekFromNow();
+        case CONST.CUSTOM_STATUS_TYPES.NEVER:
+            return CONST.CUSTOM_STATUS_TYPES.NEVER;
+        default:
+            return '';
+    }
+}
+
+/**
+ * @param {string} data - one of the values from CONST.CUSTOM_STATUS_TYPES or data in format YYYY-MM-DD HH:mm
+ * @returns {string} example: 2023-05-16 11:10 PM or 'Today'
+ */
+function getDateForTitleBasedFromType(data) {
+    const {translateLocal} = Localize;
+    switch (data) {
```

Comment:
I think we might be missing some cases here?

Reply:
originally we had such explanations of the periods
 - Never - The status will never expire. Which means it will never clear automatically.
 - After 30 minutes
 - After 1 hour
 - After today - Once a new day starts in the user’s timezone the status will be cleared automatically.
 - After a week - Same as today, but after the time of seven days.
 - Custom - You can define a custom point in time when the status will clear. See the following

In the database, we save it like 
`status: {emojiCode: '', statusText: '', clearAfter: 2022-10-10 12:12}`
After saving this date, we can distinguish only `never`, `after today`, and `custom` periods.

 So that's why this function looks so thin.